backupccl: use same mechanism for region checks as the sql package

otan opened the pull request:
This means we have to change less code when we change how region
validity works (e.g. for multi-tenant settings).

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/67521)
<!-- Reviewable:end -->

otan committed:
backupccl: use same mechanism for region checks as the sql package

This means we have to change less code when we change how region
validity works (e.g. for multi-tenant settings).

Release note: None

pbardea reviewed (approved):
Thank you!

otan commented:
thanks!

bors r=pbardea

otan commented:
oops, pushed a branch funny

bors r=pbardea
